migrations/20250908132500-encode-metadatakeys-datasets.js

```diff
@@ -0,0 +1,46 @@
+module.exports = {
+  async up(db, client) {
+    function encodeKey(key) {
+      return encodeURIComponent(key).replace(/\./g, "%2E");
+    }
+
+    function encodeMetadataKeys(metadata) {
```

Comment:
I think you can import functions from the node app here, see the history migration